File: 02-week-2--recursion/01-day-1--recursive-counting/ruby/recursive_count.rb
def recursive_count(num = 0)
  # type your code in here
  puts num
  if num < 9
    recursive_count(num + 1)
  else
    nil
  end
end

if __FILE__ == $PROGRAM_NAME
  recursive_count
end

# OPTIONAL
# Please add your pseudocode to this file
=begin
  -print num
  -if num is less than max count, call recursive_count(num + 1)
  -else return nil
=end
# And a written explanation of your solution
=begin
  I printed num and then checked to see if the value was the maximum count yet
  if it wasn't I incremented num and called the method again, if it was I simply returned nil
=end

File: 02-week-2--recursion/01-day-1--recursive-counting/javascript/recursive_count.js
function recursiveCount(num = 0) {
  // type your code here
  const addCount = (n) => {
    if (n < 9) {
      console.log(n)
      addCount(n + 1)
    } else {
      console.log(n)
      return n
    }
    return n
  }

  addCount(num)
}

if (require.main === module) {
  recursiveCount();
}

module.exports = recursiveCount;

// OPTIONAL
// Please add your pseudocode to this file
// And a written explanation of your solution
